lib/rouge/lexers/msgtrans.rb

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class MsgTrans < RegexLexer
+      title "MessageTrans"
+      desc "RISC OS message translator messages file"
+      tag 'msgtrans'
+      filenames 'Messages*'
```

Comment:
This could inadvertently match files that just happen to start with the term `Messages`. Is there a more narrow glob that could be used?

Reply:
I reckon I can safely narrrow it down to `Message(?:s|\d{1,3})` - it can have an up-to-3-digit suffix, and I didn't realise before that the `s` is dropped if the suffix is added. (This is a 27-year-old format, and there were concerns about the filename length when it was invented!) Is that still too broad?

Reply:
Hmmm. I think maybe it is. It's not a necessity for a lexer that it have a filename associated with it. I'd suggest dropping it in this case. The people who want to highlight this syntax presumably know exactly that they want to do that and can set things up appropriately.

Reply:
The other problem is that I'm not sure filename globs get parsed as regular expressions (which is why the tests are failing, I think). Another reason to remove the filename declaration.

Reply:
It's a filename glob. We use

``` ruby
File.fnmatch?(pattern, path, File::FNM_DOTMATCH | File::FNM_CASEFOLD)
```

See https://apidock.com/ruby/File/fnmatch/class for documentation of the `fnmatch?` method.

Reply:
Thanks for the pointer. As it happens, between when I wrote my comment above and when I pushed 769d9d2, I'd figured out by trial and error that it was being treated as a filename glob and that it accepted character ranges. I just forgot to update `spec/lexers/msgtrans_spec.rb` - mostly, I think, because I habitually test using `rougify` and that doesn't care what that file contains.

I don't think I really appreciated that it was a test spec before. The change in 
3e5a259  appears to be enough to let it pass CI.

I'd rather keep some sort of filename check if I can, because I have a GitLab server with 630 of these files across 560 projects, and it apppears to rely on rouge's filename guessing to identify the format. I believe GitLab can be configured to override rouge's guesses, but it would be quite fiddly to do, and likely to become stale as new files are added in future.

Reply:
I don't mind registering the `Message[\d]{1,3}` format filenames but I'm still uncomfortable about `Messages`. That just seems too generic, especially—and I'm sorry if this sounds unfair—for what appears to be a really niche language.

I'm also not sure that adding custom highlighting is as difficult as you feared. @jneen actually created the facility for this in GitLab ([documentation](https://docs.gitlab.com/ce/user/project/highlighting.html)) and it just requires adding some lines to the repo's `.gitattributes` file.

How does that seem as a compromise?

Reply:
Guess which filename the vast majority of the files use in practice? Yes, `Messages` :(

`.gitattributes` is what I was thinking of - I was kinda hoping to avoid having to create or update hundreds of those :)

How hard would it be to extend the guess-by-filename system to allow filtering according to the path leading up to the filename? The bulk of the files I'm concerned with are at `Resources/[...]/Messages` within their respective projects (i.e. zero or more subdirectories down from a directory called `Resources`).

Reply:
OK, I've taken a different approach. Rouge has the ability to disambiguate between files with the same glob that are registered to different lexers. It does this based on the content of the file and a test in `lib/rouge/guessers/disambiguation.rb`.

I've just pushed a commit to your branch that registers the `Messages` glob to the PlainText lexer and adds a disambiguation. Please feel free to suggest something better but the simplest I could see was checking whether there are any lines there begin with `\w+:\w+`. This can of course cause false positives if there does happen to be a non-MessageTrans file that has a line that begins with text that matches that pattern but I think that's sufficiently unlikely that we don't need to worry about it. 

What do you think?

Reply:
Oh, and matching a fuller path isn't a terrible idea but it: (a) requires rearchitecting part of the innards of Rouge; and (b) feels brittle. The disambiguation approach is one we take from time to time so while it is a special case in one sense, it's not so special that we'd be setting it up entirely for this lexer.

Reply:
I've been doing some analysis on the files on my GitLab server:

Matching only on filename `Messages` has two false positive matches on what are actually plaintext documentation files.

Matching on filename `Messages` under a directory `Resources` has no false positives, but 28 false negatives (files not detected as MessageTrans files when they actually are).

Matching on filename `Messages` where the file contents match your search pattern has no false positives but 15 false negatives, made up as follows:
* 7 files containing only comment lines
* 3 empty files
* 2 files containing only CJK text to the right of the colon
* 2 files where every token name contains a character not matched by `\w` (specifically, `|` or `.`)
* 1 file where every value starts with a `%`

If we could combine ( filename matches AND (contents match OR directory matches)) then it would be perfect, but in the absence of that, I think your solution is probably best.

Reply:
Given this lexer hasn't seen general use yet, I'd like to roll it out with the disambiguation rule for now and see how it goes. Speaking of which, I've tweaked the rule a little to fix some of those false negatives.